go/src/koding/klient/machine/machinegroup/aliases/aliases.go

```diff
@@ -0,0 +1,157 @@
+package aliases
+
+import (
+	"math/rand"
+	"strconv"
+	"sync"
+
+	"koding/klient/machine"
+)
+
+// colors defines the available colors for aliases. The first letter of each
+// color name must be unique across the slice.
+var colors = [...]string{
+	"aqua",
+	"black",
+	"fuchsia",
+	"green",
+	"lime",
+	"maroon",
+	"navy",
+	"olive",
+	"purple",
+	"red",
+	"silver",
+	"teal",
+	"white",
+	"yellow",
+}
+
+// fruits defines the available fruits for aliases. The first letter of each
+// fruit name must be unique across the slice.
+var fruits = [...]string{
+	"apple",
+	"banana",
+	"coconut",
+	"date",
+	"fig",
+	"grape",
+	"jackfruit",
+	"kiwi",
+	"lemon",
+	"mango",
+	"nectarine",
+	"orange",
+	"peach",
+	"quince",
+	"raisin",
+	"squash",
+	"tomato",
+	"watermelon",
+}
+
+// Aliases store alternative names for the group of machines.
+type Aliases struct {
+	mu sync.RWMutex
+	m  map[machine.ID]string
+}
+
+// New creates an empty Aliases object.
+func New() *Aliases {
+	return &Aliases{
+		m: make(map[machine.ID]string),
+	}
+}
+
+// Add binds custom alias to provided machine.
+func (a *Aliases) Add(id machine.ID, alias string) error {
+	a.mu.Lock()
+	defer a.mu.Unlock()
+	a.m[id] = alias
+
+	return nil
+}
+
+// Create generates a new alias for provided machine ID. If alias already
+// exists, it will not be regenerated.
+func (a *Aliases) Create(id machine.ID) (string, error) {
+	a.mu.Lock()
+	defer a.mu.Unlock()
+
+	if alias, ok := a.m[id]; ok {
+		return alias, nil
+	}
+
+	// Get current aliases.
+	current := make(map[string]struct{}, len(a.m))
+	for _, alias := range a.m {
+		current[alias] = struct{}{}
+	}
+
+	// Generate new alias.
+	for i := 0; ; i++ {
```

Comment:
:)

I did not comment just to comment. I have written the similar code 3 years ago, thinking i dont need a limit because it will return, right? then someone added two lines in between my code and we had infinite loop. At least you have test.


> If we have a truly uniform random number generator

Do we have? :)